src/server/raftkv/mod.rs

```diff
@@ -879,6 +885,32 @@ impl ReadIndexObserver for ReplicaReadLockChecker {
                         .observe(begin_instant.saturating_elapsed().as_secs_f64());
                 }
             }
+            if rctx.locked.is_none() {
+                if let (Some(region_start_key), Some(region_end_key)) =
+                    (region_start_key, region_end_key)
+                {
+                    // check if there is a memory lock on a region
+                    let start_key = key_bound(region_start_key.to_vec());
+                    let end_key = key_bound(region_end_key.to_vec());
+
+                    let res = self.concurrency_manager.read_range_check(
```

Comment:
For scenarios with a high number of follower read requests and frequent writes, checking the entire region's memory lock for conflicts on each follower read request may increase contention, leading to performance degradation in such cases. The strategy here may need further consideration.

From a product perspective, if there is no particularly suitable default enablement strategy, adding an optimization switch is one option. However, this comes at the cost of increased usage complexity.

Reply:
`IsolationLevel::RcCheckTs` is used by another feature, mixing usage may achive the goal but will expand its semantic scope. Or adding another new constant to achive this.